### ISSUE TITLE
Empty hint bar appears on bottom of screen

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/firefox-tv#2133]
## Steps to reproduce
- Open debug build
- Loda a page

### Expected behavior
No hint bar or hint bar with content

### Actual behavior
Hint bar appears but is empty

### Device information
* Fire TV device: Cube
* Latest affected Firefox version: master
* Earliest affected Firefox version: ?

~I think this is  a regression from https://github.com/mozilla-mobile/firefox-tv/issues/2029 (which just landed).~
EDIT: regression from #2118

**Comment by athomasmoz** [comment]
@mcomella is this only on debug builds?

**Comment by Baron-Severin** [comment]
@athomasmoz 
> mcomella is this only on debug builds?

It's on both release and master.  We checked and it's not reproducible on the version we're currently pushing out to users. We also have a PR with a fix open, just waiting on review.

**Comment by athomasmoz** [comment]
Yikes! Thanks for jumping in to fix this. I'm going to add it to the sprint - can you throw an estimate on?

**Comment by mcomella** [comment]
There are three underlying problems here:
1. The hint bar, as is, is broken. See #2138 for the cause and partial fix. This is a regression from #2118.
1. We need to figure out how to prevent errors like this (disabled A/B test appears on the screen) from happening again: e.g. adding another absolute "A/B" test state or adding another layer between the A/B test code and the on/off switch (e.g. Netflix architecture).
1. The hint bar code is confusing because it is not modeling the animation state (e.g. as absolute position), making animations a side effect of our functional architecture

We need to address 1 and could also try to address 2. I don't think we should address 3 (it's mostly working now, no point in adding risk for something that won't change often). See my review for more info on 1-3: https://github.com/mozilla-mobile/firefox-tv/pull/2138#pullrequestreview-232324982

**Comment by dnarcese** [comment]
#1 is addressed in #2148. Will leave this issue open if we want to address the other items later.

**Comment by mcomella** [comment]
@dnarcese Can you file a separate issue for 2? I feel it'd be better to group these discussions separately. I don't care to file an issue for 3 (or other similar aspirational refactors: if they're truly important, they'll come up again as part of a different bug; when they're filed, they usually end up unprioritized and forgotten about).
